### PR TITLE
[www/docs] add classmethods,staticmethod,data,attribute,function,class explicitly

### PR DESCRIPTION
**Pull request by akotlar**
The portions I committed after your initial approval weren't merged. That almost seems like a CI bug.

edit: those additional changes apparently didn't get pushed. This fixes.

This expands on the work that you approved, by adding the remaining sections. There are still only 2 style groups, unfortunately split between 6 classes.

fyi @cseed (styling)

### CODE REVIEW

**Comment by akotlar** [comment]
Now anything that looks like:

Screen Shot 2019-10-12 at 3 48 03 PM

Also gets proper offset.

I ripgrep'd the directory to look for any other `dl` classes. I think this covers everything

**Comment by akotlar** [comment]
Also, the rules are as specific as they are to avoid using `!important` tags. These rules have slightly higher specificity than the rules they're overriding.

File: hail/python/hail/docs/_static/rtd_modifications.css [hunk 1]
```diff
@@ -2,24 +2,28 @@
...
+  margin-top: -50px;
+  padding-top: 50px;
 }
 
-.section:before {
-  height: 51px;
-  content: "";
-  display: block;
+dl+dl {
+  margin-top: 50px;
```
Comment (akotlar) [inline hail/python/hail/docs/_static/rtd_modifications.css]: this is a quick fix to the problem that some of these focusable blocks have no documentation, and therefore the space between them is smaller than the height needed for the offset, making them unclickable.

Comment (akotlar) [reply hail/python/hail/docs/_static/rtd_modifications.css]: ex: https://hail.is/docs/0.2/types.html#hail.expr.types.HailType.subst

Comment (akotlar) [reply hail/python/hail/docs/_static/rtd_modifications.css]: If we document each function I don't think the problem will emerge.

File: hail/python/hail/docs/_static/rtd_modifications.css [hunk 2]
```diff
@@ -130,7 +127,7 @@ div.nboutput div.output_area.stderr {
 
 .bolditalic {
   font-weight: bold;
-  font-italic: italic;
+  font-style: italic;
```
Comment (akotlar) [inline hail/python/hail/docs/_static/rtd_modifications.css]: too small for a separate PR I think. a mistake I noticed while making the present update.

File: hail/python/hail/docs/_static/rtd_modifications.css [hunk 3]
```diff
@@ -29,16 +38,22 @@
...
-.rst-content dl:not(.docutils) dl.attribute dt:first-child {
-  border-top: 50px solid transparent;
-  margin-top: -50px;
+.rst-content dl:not(.docutils) dl.attribute dt:first-child,
+.rst-content dl:not(.docutils) dl.classmethod dt,
+.rst-content dl:not(.docutils) dl.classmethod dt:first-child,
+.rst-content dl:not(.docutils) dl.staticmethod dt,
+.rst-content dl:not(.docutils) dl.staticmethod dt:first-child {
+  border-top: 49px solid transparent;
+  margin-top: -49px;
```
Comment (akotlar) [inline hail/python/hail/docs/_static/rtd_modifications.css]: this makes it flush with header

File: hail/python/hail/docs/_static/rtd_modifications.css (hunk 1 above)
Comment (tpoterba) [reply hail/python/hail/docs/_static/rtd_modifications.css]: that one shouldn't be documented at all.

Comment (akotlar) [reply hail/python/hail/docs/_static/rtd_modifications.css]: Should we have functions in our documentation that have no explanation?

Comment (akotlar) [reply hail/python/hail/docs/_static/rtd_modifications.css]: If you meant that even the function links shouldn’t be present I agree. Will make an issue

Comment (akotlar) [reply hail/python/hail/docs/_static/rtd_modifications.css]: #7292

**Pull request merged**